clang/test/SemaTemplate/cwg2398.cpp

```diff
@@ -326,6 +320,48 @@ namespace classes {
       // expected-error@-1 {{no matching function for call}}
     }
   } // namespace packs
+  namespace nested {
```

Comment:
Do we need more tests?
Maybe add the example from #100692 ?

Reply:
The test case from 100692 doesn't have anything interesting new on top of this one. There might be differences in the kinds of template arguments which have default deductions, but we already test all kinds when matching TTP to  TTP, so I am not sure it would be worth doing a power set where we test every combination of matching with every kind.